tests/profile/update/loginWithUpdatedPassword.spec.js

```
@@ -1,29 +1,30 @@
-import { test } from '../../_fixtures/fixtures';
-import { signUpUser } from '../../../src/ui/actions/auth/signUpUser';
+import { test, expect } from '../../_fixtures/fixtures';
+import { InternalHomePage } from '../../../src/ui/pages/home/InternalHomePage';
+import { SignInPage } from '../../../src/ui/pages/auth/SignInPage';
 
-let newPassword;
+test('Login with new password after it was updated via API', async ({
+  page,
+  registeredUser,
+  factories,
+  usersApi,
+}) => {
+  const signInPage = new SignInPage(page);
+  const internalHomePage = new InternalHomePage(page);
 
-test.beforeEach(async ({ page, user, factories }) => {
-  await signUpUser(page, user);
+  // Must use registeredUser (not loggedInUserAndPage) so the test performs a real UI login from a logged-out state.
+  // page fixture starts with no active session.
 
-  newPassword = factories.user.generatePassword();
-});
+  const newPassword = factories.user.generatePassword();
+
+  const updatedUser = { ...registeredUser, password: newPassword };
+
+  const updateResponse = await usersApi.updateUser(updatedUser);
+  expect(updateResponse.ok()).toBeTruthy();
 
-test('Login with new password after it was updated from settings', async ({
-  editSettingsPage,
-  viewUserProfilePage,
-  signInPage,
-  internalHomePage,
-  user,
-}) => {
-  await editSettingsPage.open();
-  await editSettingsPage.fillNewPasswordField(newPassword);
-  await editSettingsPage.clickUpdateSettingsButton();
-  await viewUserProfilePage.clickEditProfileSettingsLink();
-  await editSettingsPage.clickLogoutButton();
   await signInPage.open();
```

Add an explicit assertion that the sign-in page/form is visible before filling fields (for example `signInPage.assertFormVisible()` or similar) to ensure you are operating from a logged-out state and to fail fast if navigation didn't land on the expected page.